src/EventSourcingDb/Client.cs

```diff
@@ -500,23 +525,30 @@ public async IAsyncEnumerable<JsonElement> RunEventQlQueryAsync(
         }
     }
 
-    public async IAsyncEnumerable<TRow> RunEventQlQueryAsync<TRow>(
-        string query,
-        [EnumeratorCancellation] CancellationToken token = default)
+    private TRow DeserializeRow<TRow>(JsonElement payload)
     {
-        await foreach (var jsonElement in RunEventQlQueryAsync(query, token))
+        TRow? row;
+        try
         {
-            TRow? row;
-            try
+            if (typeof(TRow) == typeof(Event))
```

Comment:
Wir versuchen in der Regel, `if`/`else`-Konstrukte zu vermeiden. Stattdessen prüfen wir auf das Gegenteil (da es sich hier um eine Einzeile handelt) und verwenden `continue`. Auf diese Weise kannst Du den `else`-Zweig komplett weglassen und sparst Dir eine zusätzliche Einrückungsebene.